### PR TITLE
feat: cleaner ISS hanlder wiring

### LINKED ISSUE #13456: Clean up ISS handler wiring
Clean up the wiring for the ISS handler.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13456

### CODE REVIEW

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 1]
```diff
@@ -846,6 +850,45 @@ public IssDetector buildIssDetector() {
...
+     * Provide an ISS handler in place of the platform's default ISS handler.
+     *
+     * @param issHandler the ISS handler to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withIssHandler(@NonNull final IssHandler issHandler) {
+        throwIfAlreadyUsed();
+        if (this.issHandler != null) {
+            throw new IllegalStateException("ISS handler has already been set");
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:863]: Is that really a problem?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            return false;
+        });
+    }
+
+    /**
+     * This method is called when there is a self ISS.
+     *
+     * @param round the round of the ISS
+     */
+    private void selfIssObserver(@NonNull final Long round) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:112]:
```suggestion
    private void selfIssObserved(@NonNull final Long round) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            case SELF_ISS -> selfIssObserver(issNotification.getRound());
+            case OTHER_ISS -> otherIss();
+            case CATASTROPHIC_ISS -> catastrophicIssObserver(issNotification.getRound());
+        }
+    }
+
+    /**
+     * This method is called whenever any node is observed in disagreement with the consensus hash.
+     */
+    private void otherIss() {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:75]:
```suggestion
    private void otherIssObserved() {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+import com.swirlds.platform.system.SystemExitCode;
+import com.swirlds.platform.system.state.notifications.IssNotification;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * A standard implementation of {@link IssHandler}.
+ */
+public class DefaultIssHandler implements IssHandler {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:33]: If we move this class directly in a `com.swirlds.platform.state.iss.internal` package we can split better between public and private API

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    /**
+     * This method is called when there is a self ISS.
+     *
+     * @param round the round of the ISS
+     */
+    private void selfIssObserver(@NonNull final Long round) {
+
+        if (halted) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:114]: Can we ignore concurrency here or should halted be refactored to be atomic?

**Review by hendrikebbers — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:112]: renamed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:33]: Moved, although I should point out that the current refactoring effort is not attempted to address the package structure for platform api/impl. Right now package structure is very ad hoc, and I don't think we have a clear plan for how we want to structure things yet.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java (hunk 5 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java:114]: Concurrency is intentionally ignored at this scope.

The ISS handler is a direct scheduler (meaning it is executed on the caller's thread), and the caller is a sequential scheduler (meaning there is a single logical thread). 

The goal of the wiring framework is to make business logic like this entirely devoid of threading primitives, and to handle all threading concerns at the wiring framework level. In current platform business logic, if you see things like atomic variables, locks, or the `synchronized` keyword, it usually means it is legacy code that we haven't gotten around to refactoring.

**Review by alittley — APPROVED** [review]

**Pull request merged**